### ISSUE TITLE
[Feature Request] Python - like behavior of tf.string_split. Consider whole multi-byte delimiter.

### ISSUE THREAD

**Issue by nxphi47** [tensorflow/tensorflow#18271]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS, Ubuntu
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:1.7
- **Python version**: 3.6,3.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:No
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem

`tf.string_split` is documented as `If delimiter contains multiple bytes, it is treated as a set of delimiters with each considered a potential split point.` Hence, each character in a multi-byte string delimiter will become a separate delimiter, which is very non-intuitive and difficult to implement.

so for example, I want to split by to the token `<eos>`
```
tf.string_split(["hello world <eos> tensorflow"], delimiter="<eos>").values
# equal:     ['h', 'll', ' w', 'rld ', ' t', 'n', 'rfl', 'w']  
# should be ['hello world ', ' tensorflow]  
```

Can tf.string_split be reimplemented so that it consider the whole delimiter string, whether single or multiple character, to be its true delimiter instead making every single character in it delimiter?

Thank you

**Comment by Luonic** [comment]
Also it would be very nice to be able to split utf-8 encoded string into multibyte characters instead of separate raw bytes to be able to implement character-level NLP completely with TF's OPs. Current solution is dirty hack: py_func that decodes string to utf-8, insert some separator (for example: "|") after each character and then this string tensor is separated by tf.string_split(input_tensor, delimiter="|")
It could be better)

**Comment by bignamehyp** [comment]
@martinwicke I think this is a good feature suggestion. What do you think?

**Comment by martinwicke** [comment]
I agree `string_split` should work exactly like Python's [`str.split`](https://docs.python.org/2/library/stdtypes.html#str.split). This would be a fantastic feature to contribute, but we'll have to talk details about backwards compatibility, and we need to make sure this works in an unsurprisingly way across Python 2 and 3 and different string encodings.

**Comment by nxphi47** [comment]
I am not sure if we actually need to compute backward pass with strings. Usually people deal with strings at preprocess steps, which there is no need to backpropagate. 

Yet, is possible to backpropagate with strings any way, even with the current tf.string_split implementation ?

**Comment by martinwicke** [comment]
I'm not talking about gradients, I'm talking about backwards compatibility guarantees for the API. We cannot change the current behavior (the delimiter string will be unpacked into a set of single characters). We have to find a backwards compatible version for the signature of string_split does doesn't change the current behavior.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by yongtang** [comment]
Added a PR #19650 to expose in `tf.strings.split` which matches the behaviors of python's `str.split`.
